editor/editor_style.go

```diff
@@ -98,10 +114,90 @@ func (e EditorStyle) Layout(gtx layout.Context) layout.Dimensions {
 	}
 	e.Editor.LineHeight = e.LineHeight
 	e.Editor.LineHeightScale = e.LineHeightScale
-	dims = e.Editor.Layout(gtx, e.shaper, e.Font, e.TextSize, textColor, selectionColor)
-	if e.Editor.Len() == 0 {
-		call.Add(gtx.Ops)
+
+	if !e.ShowLineNum {
+		d := e.Editor.Layout(gtx, e.shaper, e.Font, e.TextSize, textColor, selectionColor)
+		if e.Editor.Len() == 0 {
+			call.Add(gtx.Ops)
+		}
+		return d
+	}
+
+	dims = layout.Flex{
+		Axis: layout.Horizontal,
+	}.Layout(gtx,
+		layout.Rigid(func(gtx layout.Context) layout.Dimensions {
+			lines, _ := e.Editor.VisibleLines()
+			return lineNumberBar{
+				shaper:          e.shaper,
+				lineHeight:      e.LineHeight,
+				lineHeightScale: e.LineHeightScale,
+				color:           misc.WithAlpha(e.Color, 0xb6),
+				typeFace:        e.Font.Typeface,
+				textSize:        e.TextSize,
+				positions:       lines,
+			}.Layout(gtx)
+		}),
+
+		layout.Rigid(layout.Spacer{Width: unit.Dp(20)}.Layout),
```

Comment:
would be nice to add a border of 1px on right side of the line number bar.

Reply:
I think the two styles are acceptable, but I guess it would be better for the line number bar to provide some customization, and this styling issue can be solved by adding more options. What do you think?

Reply:
Totally agree 👍 it would be nice to be able to customise it